filebeat/harvester/harvester.go

```diff
@@ -17,12 +17,15 @@ import (
 	"fmt"
 	"regexp"
 
+	"github.com/satori/go.uuid"
+
 	"github.com/elastic/beats/filebeat/config"
 	"github.com/elastic/beats/filebeat/harvester/encoding"
 	"github.com/elastic/beats/filebeat/input"
 )
 
 type Harvester struct {
+	Id                 uuid.UUID
```

Comment:
I now realise I had one more idea somewhere in the back of my head when I added it: Stats reporting. So if stats about harvesters are reported to ES it would be always clear which harvester it is. But probably the same would be true for selecti `beat.UUID + idCount` as the idCount would be unique for all harvesters ever started in a beat.